custom-idl/index.js

```diff
@@ -14,14 +14,15 @@
 
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
-const WebIDL2 = require('webidl2');
+import fs from 'fs-extra';
+import path from 'path';
+import {fileURLToPath} from 'url';
+import {parse} from 'webidl2';
 
 // Load text (UTF-8) files from a directory and return an object mapping each
 // name (sans extension) to the parsed result of that text.
-const parseIDL = () => {
-  const files = fs.readdirSync(__dirname);
+const parseIDL = async () => {
+  const files = await fs.readdir(fileURLToPath(new URL('.', import.meta.url)));
```

Comment:
To reduce the ugly stuff with URLs, can you do `const dirname = fileURLToPath(import.meta.url)` once and then use `await fs.readdir(dir)` here?